src/main/java/org/opensearch/plugin/insights/core/listener/QueryInsightsListener.java

```diff
@@ -239,7 +240,16 @@ private void constructSearchQueryRecord(final SearchPhaseContext context, final
                 );
             }
 
-            String hashcode = QueryShapeGenerator.getShapeHashCodeAsString(request.source(), false);
+            Set<String> successfulShardIndices = searchRequestContext.getSuccessfulSearchShardIndices();
+            if (successfulShardIndices == null) {
+                successfulShardIndices = Collections.emptySet();
+            }
+            String hashcode = QueryShapeGenerator.getShapeHashCodeAsString(
```

Comment:
We should have a knob to enable/disable field type in the query shape?

Reply:
Or do we plan to show this if fieldName enabled?

Reply:
Do we want separate knobs for field name and field type?

Reply:
I did not think of a good way to handle this. A new method param for each field data is not sustainable. Maybe a list of string constants or enum would work best. 

Adding knobs is not complex. Given that showFields is currently disabled, I vote we implement knobs when we decide to turn on showFields. This way we can tailor it directly for our needs and not spend extra effort now.